### DISCUSSION TITLE
Does PatchOperation is ACID ?

### DISCUSSION THREAD

**Discussion by panagushin** [ravendb/ravendb#14632] [Q&A] [answered]
`According` to this [page](https://ravendb.net/docs/article-page/5.3/csharp/client-api/faq/transaction-support) PatchOperation is ACID, is it? Or I misunderstand something

I mean, when I do something like this, can I suppose it will execute in ACID manner?

```
var patch = new PatchOperation(
  Id,
  null,
  new PatchRequest
  {
    Script,
    Values
  },
  new PatchRequest
  {
    Script,
    Values
  }
);

var result = await _store.Operations.SendAsync(patch);
```

**Comment by ayende** [comment, marked as answer]
Yes, this is absolutely transactional.

**Reply by ayende** [reply]
Note that _update by query_ isn't transactional for the whole operation

**Reply by ayende** [reply]
But that a Patch operation is running on specific documents and is transactional across them.
Also can be combined with other operations in a transactional manner.

**Reply by panagushin** [reply]
thanks
